hw/ip/otbn/dv/rig/rig/model.py

```diff
@@ -779,6 +779,94 @@ def update_for_bnmovr(self, prog_insn: ProgInsn) -> None:
         elif grs_inc:
             self._inc_gpr(grs, grs_val, 1, 31)
 
+    def update_for_bnxor(self, prog_insn: ProgInsn) -> None:
+        '''Update model state after an BN.XOR
+
+        If the source register happens to have a known value, we can do the
+        addition and store the known result.
+
+        '''
+        insn = prog_insn.insn
+        op_vals = prog_insn.operands
+        assert insn.mnemonic == 'bn.xor'
+        assert len(insn.operands) == len(op_vals)
+
+        exp_shape = (isinstance(insn.operands[0].op_type, RegOperandType) and
+                     insn.operands[0].op_type.reg_type == 'wdr' and
+                     insn.operands[0].op_type.is_dest() and
+                     isinstance(insn.operands[1].op_type, RegOperandType) and
+                     insn.operands[1].op_type.reg_type == 'wdr' and
+                     not insn.operands[1].op_type.is_dest() and
+                     isinstance(insn.operands[2].op_type, RegOperandType) and
+                     insn.operands[2].op_type.reg_type == 'wdr' and
+                     not insn.operands[2].op_type.is_dest() and
+                     isinstance(insn.operands[4].op_type, ImmOperandType))
+        if not exp_shape:
+            raise RuntimeError('BN.XOR instruction read from insns.yml is '
+                               'not the shape expected by '
+                               'Model.update_for_bnxor.')
+
+        wrs1_val = self.get_reg('wdr', op_vals[1])
+        wrs2_val = self.get_reg('wdr', op_vals[2])
+
+        result = None
+
+        if (op_vals[1] == op_vals[2]) and (op_vals[4] == 0):
+            result = 0
+        elif wrs1_val is None or wrs2_val is None:
+            pass
+        else:
+            if op_vals[3]:
+                src2_val = wrs2_val >> op_vals[4]
+                result = (wrs1_val ^ src2_val)
+            else:
+                src2_val = wrs2_val << op_vals[4]
+                result = (wrs1_val ^ src2_val)
```

Comment:
This is fine, but I'd probably drop it since we don't technically need it. No particularly strong feelings either way, though.